pymeasure/instruments/keithley/keithley2510.py

```diff
@@ -0,0 +1,207 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2025 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from time import sleep, time
+
+import numpy as np
+
+from pymeasure.instruments import Instrument, SCPIMixin
+from pymeasure.instruments.validators import truncated_range, strict_discrete_set
+
+
+class Keithley2510(SCPIMixin, Instrument):
+    """Represents the Keithley 2510 TEC Sourcemeter and provides a high-level interface for
+    interacting with the instrument.
+    """
+
+    def __init__(self, adapter, name="Keithley 2510 TEC SourceMeter", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+
+    ###########
+    # Control #
+    ###########
+
+    source_enabled = Instrument.control(
+        "OUTPut?",
+        "OUTPut %d",
+        """Control whether the source is enabled (bool).
+        The convenience methods :meth:`~.Keithley2510.enable_source`
+        and :meth:`~.Keithley2510.disable_source` can also be used.""",
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+
+    temperature_setpoint = Instrument.control(
+        ":SOURce:TEMPerature?",
+        ":SOURce:TEMPerature %g",
+        """Control the temperature setpoint, in degrees centigrade.
+        Takes values from -50 to 225.""",
+        validator=truncated_range,
```

Comment:
We typically do not recommend truncated validators: The user might set the value to -60, believing, that it will be set to -60, as there is no error.

Typically, it is better to use the strict version, such that a user will notice, that the command did not execute as expected.